spec/models/submission_spec.rb

```diff
@@ -2,6 +2,40 @@
 
 RSpec.describe Submission do
 
+  it { should validate_presence_of(:user) }
+  it { should validate_presence_of(:submission_type) }
+
+  describe '#submission_type' do
+    let(:submission) { create(:submission) }
+
+    it 'allows only certain submission type values' do
+      %w(population traits qtl linkage_map).each do |t|
+        submission.submission_type = t
+        expect(submission.save).to be_truthy
+        submission.reload
+        expect(submission.submission_type).to eq t
+        expect(submission.send(t+'?')).to be_truthy
+      end
+      expect { submission.submission_type = 'wrong_submission_type' }.
+        to raise_error ArgumentError
+    end
+
+    it 'honors symbols as type values' do
+      submission.submission_type = :population
+      expect(submission.save).to be_truthy
+    end
+
+    it 'provides handy scopes to query certain types' do
```

Comment:
Some say it's superfluous but I believe otherwise. Because with time you want to upgrade AR as well.

But in this specific case I also check that no one modifies the four submission enum to sth else. I prefer it stays that way.